File: app/src/main/java/com/usher/demo/base/BaseActivity.java
package com.usher.demo.base;

import android.annotation.SuppressLint;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;

import com.usher.demo.utils.PermissionUtil;
import com.usher.demo.utils.RxUtil;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import io.reactivex.Observable;

@SuppressLint("Registered")
public class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        View decorView = getWindow().getDecorView();
        int uiOptions = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
        decorView.setSystemUiVisibility(uiOptions);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        Observable.range(0, grantResults.length)
                .map(i -> grantResults[i] == PackageManager.PERMISSION_GRANTED)
                .reduce(true, (acc, curr) -> acc && curr)
                .as(RxUtil.autoDispose(this))
                .subscribe(granted -> {
                    if (granted) {
                        PermissionUtil.grant();
                    } else {
                        PermissionUtil.deny();
                    }
                });
    }
}

File: app/src/main/java/com/usher/demo/base/BaseActivity.kt
package com.usher.demo.base

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatDelegate
import com.twigcodes.ui.activity.BaseActivity
import com.usher.demo.main.DemoListActivity
import com.usher.demo.utils.CommonUtil

@SuppressLint("Registered")
open class BaseActivity(contentLayoutId: Int, statusBarThemeForDayMode: Theme = Theme.LIGHT_AUTO, fullScreen: Boolean = true, hideNavigationBar: Boolean = false) : BaseActivity(contentLayoutId, statusBarThemeForDayMode, fullScreen, hideNavigationBar) {
    enum class NightModeStrategy {
        AUTO,
        YES,
        NO
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val nightMode = when (CommonUtil.getNightModeStrategy(this)) {
            NightModeStrategy.AUTO -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
            NightModeStrategy.YES -> AppCompatDelegate.MODE_NIGHT_YES
            NightModeStrategy.NO -> AppCompatDelegate.MODE_NIGHT_NO
        }

        AppCompatDelegate.setDefaultNightMode(nightMode)
    }

    override fun onUIModeChanged(isNightMode: Boolean) {
        startActivity(Intent(this, DemoListActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
        })
//        updateStatusBarTheme(isNightMode)
    }
}